### ISSUE TITLE
Red Moon dims outside of scheduled times

### ISSUE THREAD

**Issue by consideredwords** [LibreShift/red-moon#194]
I downloaded 3.0.0 from F-Droid and it ran fine. Since I upgraded it via F-Droid to 3.2.0 I've found that Red Moon won't switch off outside its scheduled times. I've checked the times (based on location and timer) and say it's meant to end at 06:53, it's still red and dim at 10:00. 

I've uninstalled, rebooted, reinstalled (through F-Droid and android), rebooted and repeated with version 3.0.0. 

The other bug is that the widget icon has disappeared too and instead it's just a grey block. Not as crippling but a bit annoying.

Running Blackberry Keyone with Android 7.1.1, I installed a security patch (5-Sep-17) earlier this week before the upgrade. It has permission to draw over other apps etc so it works in that it goes red, it just won't stop.

**Comment by smichel17** [comment]
Hi, thanks for the report. Are you able to pause the filter manually from the notification, or does it not go off at all?

**Comment by consideredwords** [comment]
Thanks, if I flick the on/off switch for Red Moon it works (same with the widget), and turning it back on again will bring the red dim back.

**Comment by consideredwords** [comment]
I should add I downloaded the debug version from your repository (which I now can't find, it was the September 17 version with debug in the file title)

**Comment by smichel17** [comment]
@consideredwords When Red Moon turns on, we set the schedule for the following night. When it turns off, we set the schedule for the next morning. In both cases, we only ever set the schedule for the next day.

I believe that a package structure change I made in v3.2.0 caused events scheduled by earlier versions not to be triggered, and therefore never re-scheduled for the following day (although, I'm not sure why that would not affect both ends of the schedule).

If I'm right about that, turning the schedule on and off should force a refresh and permanently fix the issue. If that's not the case, let me know and I'll re-open this issue.
